packages/pymedphys_dicom/src/pymedphys_dicom/dicom/dose.py

```diff
@@ -32,100 +30,148 @@
 
     if reshape:
-        warnings.warn((
-            '`dose_from_dataset` currently reshapes the dose grid. In a '
-            'future version this will no longer occur. To begin using this '
-            'function without the reshape pass the parameter `reshape=False` '
-            'when calling `dose_from_dataset`.'), UserWarning)
-        pixels = np.transpose(
-            ds.pixel_array, (1, 2, 0))
-    else:
-        pixels = ds.pixel_array
+        warnings.warn(
+            ('The `reshape` parameter no longer does anything. Please remove '
+             'this parameter. In a future version this parameter will no '
+             'longer be accepted.'), UserWarning)
 
-    dose = pixels * ds.DoseGridScaling
+    dose = ds.pixel_array * ds.DoseGridScaling
 
     return dose
 
 
-def axes_and_dose_from_dicom(dicom_filepath):
-    ds = pydicom.dcmread(dicom_filepath, force=True)
-    axes = xyz_axes_from_dataset(ds)
-    dose = dose_from_dataset(ds)
+def dicom_dose_interpolate(interp_coords, dose: pydicom.Dataset):
+    """Interpolates across a DICOM dose dataset.
+
+    Parameters
+    ----------
+    dose : pydicom.Dataset
+        An RT DICOM Dose object
+    interp_coords : tuple(z, y, x)
+        A tuple of coordinates in DICOM order, z axis first, then y, then x
+        where x, y, and z are DICOM axes.
+    """
+
+    interp_z = np.array(interp_coords[0], copy=False)[:, None, None]
+    interp_y = np.array(interp_coords[1], copy=False)[None, :, None]
+    interp_x = np.array(interp_coords[2], copy=False)[None, None, :]
+
+    coords, dose = zyx_and_dose_from_dataset(dose)
+    interpolation = RegularGridInterpolator(coords, dose)
+
+    try:
+        result = interpolation((interp_z, interp_y, interp_x))
+    except ValueError:
+        print(f"coords: {coords}")
+        raise
 
-    return axes, dose
+    return result
 
 
-def load_dicom_data(ds, depth_adjust):
-    dose = dose_from_dataset(ds)
-    crossplane, vertical, inplane = xyz_axes_from_dataset(ds)
+def depth_dose(depths, dose: pydicom.Dataset, plan: pydicom.Dataset):
+    depths = np.array(depths, copy=False)
 
-    depth = vertical + depth_adjust
+    surface_entry_point = get_surface_entry_point(plan)
+    depth_adjust = surface_entry_point.y
 
-    return inplane, crossplane, depth, dose
+    y = depths + depth_adjust
+    x, z = [surface_entry_point.x], [surface_entry_point.z]
 
+    coords = (z, y, x)
+
+    extracted_dose = np.squeeze(dicom_dose_interpolate(coords, dose))
+
+    return extracted_dose
+
+
+def profile(displacements, depth, direction, dose: pydicom.Dataset,
+            plan: pydicom.Dataset):
+
+    displacements = np.array(displacements, copy=False)
 
-def extract_depth_dose(ds, depth_adjust, averaging_distance=0):
-    inplane, crossplane, depth, dose = load_dicom_data(ds, depth_adjust)
+    surface_entry_point = get_surface_entry_point(plan)
+    depth_adjust = surface_entry_point.y
```

Comment:
I do think this is important, because physicists would routinely extract profiles for arbitrary beam angles (indeed, VMAT beams) when commissioning - e.g. for film comparison.

Reply:
Yup, fair point. Likely worth considering how to handle that in a neat way.

`depth_adjust` is the number needed to add to the supplied depth request values to arrive in DICOM coords. It assumes depth is in the y axis, so will completely break for anything other than gantry 0.

Reply:
We could include gantry angle as a parameter that defaults to 0 and do the trig?

Reply:
(not this PR)

Reply:
> We could include gantry angle as a parameter that defaults to 0 and do the trig?

Sure, makes sense.

Reply:
Pulled this out into an issue for another PR.